cmd/server/cmd/server.go

```diff
@@ -258,10 +260,16 @@ func runServer(sa *serverArgs, printf, fatalf shared.FormatFn) {
 		grpcOptions = append(grpcOptions, grpc.Creds(credentials.NewTLS(tlsConfig)))
 	}
 
+	// setup server grpc interceptors for monitoring
+	grpc_prometheus.EnableHandlingTimeHistogram()
+
 	if sa.enableTracing {
 		tracer := bt.New(tracing.IORecorder(os.Stdout))
 		ot.InitGlobalTracer(tracer)
-		grpcOptions = append(grpcOptions, grpc.UnaryInterceptor(otgrpc.OpenTracingServerInterceptor(tracer)))
+		grpcOptions = append(grpcOptions,
+			grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(otgrpc.OpenTracingServerInterceptor(tracer), grpc_prometheus.UnaryServerInterceptor)))
```

Comment:
Ug, this API is unfortunate. It'd be cool if I could call `grpc.UnaryInteceptor(grpc_middleware...)` many times and have grpc_middleware handle the chaining under the hood; I might play with that in some down time.

In the mean time can we swap things so we build up an array of interceptors-to-be-chained then register them once rather than branching and registering things in the branches? I worry it's easy to screw up accidentally in a future change atm.